pkg/autodiscover/autodiscover_podset.go

```diff
@@ -71,8 +71,10 @@ func findDeploymentByLabel(
 		if len(dps.Items) == 0 {
 			log.Warn("Did not find any deployments in ns=%s", ns)
 		}
-
 		for i := 0; i < len(dps.Items); i++ {
```

Comment:
To follow the same pattern as for pods and operator, create a findDeploymentsMatchingAtLeastOneLabel function